src/tree/updater_gpu_hist.cu

```diff
@@ -1355,17 +1380,20 @@ inline void DeviceShard<GradientSumT>::CreateHistIndices(
          (entries_d.data().get(), data_vec.data() + row_ptrs[batch_row_begin],
           n_entries * sizeof(Entry), cudaMemcpyDefault));
     const dim3 block3(32, 8, 1);  // 256 threads
-    const dim3 grid3(dh::DivRoundUp(n_rows, block3.x),
+    const dim3 grid3(dh::DivRoundUp(device_row_state.batch_n_rows, block3.x),
                      dh::DivRoundUp(row_stride, block3.y), 1);
     CompressBinEllpackKernel<<<grid3, block3>>>
         (common::CompressedBufferWriter(num_symbols),
          gidx_buffer.data(),
          row_ptrs.data().get() + batch_row_begin,
          entries_d.data().get(),
-         gidx_fvalue_map.data(), feature_segments.data(),
-         batch_row_begin, batch_nrows,
+         gidx_fvalue_map.data(),
+         feature_segments.data(),
+         device_row_state.n_rows_processed + batch_row_begin,
```

Comment:
There are too many similarly named to variables here. e.g. device_row_state.batch_n_rows, batch_nrows. I want to be very clear what each of these variables is doing through descriptive naming and comments. I don't mind if the names are long, it's more important to know what is happening.

Reply:
i'll try and come up with some names.